pkg/compactor/block_upload.go

```diff
@@ -30,12 +34,21 @@ import (
 	util_log "github.com/grafana/mimir/pkg/util/log"
 )
 
-// Name of file where we store a block's meta file while it's being uploaded.
 const (
-	uploadingMetaFilename = "uploading-meta.json"
-	validationFilename    = "validation.json"
+	uploadingMetaFilename       = "uploading-meta.json" // Name of the file that stores a block's meta file while it's being uploaded
+	validationFilename          = "validation.json"     // Name of the file that stores a heartbeat time and possibly an error message
+	validationHeartbeatInterval = 1 * time.Minute       // Duration of time between heartbeats of an in-progress block upload validation
+	validationHeartbeatTimeout  = 5 * time.Minute       // Maximum duration of time to wait until a validation is able to be restarted
 )
 
+type BlockUploadConfig struct {
+	SkipValidation bool `yaml:"skip_validation" category:"experimental"`
+}
+
+func (cfg *BlockUploadConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet, logger log.Logger) {
+	f.BoolVar(&cfg.SkipValidation, prefix+".validation-heartbeat-interval", false, "Skip validation of blocks when completing a block upload")
```

Comment:
```suggestion
	f.BoolVar(&cfg.SkipValidation, prefix+".skip-validation", false, "Skip validation of blocks when completing a block upload")
```

Reply:
Can we please rename this flag to `block-validation-enabled`? It is more consistent with other Mimir options.